api/controllers/game/ready.js

```diff
@@ -5,10 +5,11 @@ module.exports = function (req, res) {
   if (req.session.game && req.session.usr) {
     const promiseGame = gameAPI.findGame(req.session.game);
     const promiseUser = userAPI.findUser(req.session.usr);
-    Promise.all([promiseGame, promiseUser])
+    const promiseLock = sails.helpers.lockGame(req.session.game);
+    Promise.all([promiseGame, promiseUser, promiseLock])
```

Comment:
This is where the rubber meets the road for actually using the lock. It will be much cleaner once #448 is merged because the control flow is simpler and we can use async/await here, but I like how simple and minimally invasive it is to apply a lock/unlock pattern in an existing endpoint

Reply:
Interesting question. I think really the lock should come _before_ getting the game and user so that whatever game data you query for is up to date by the time you get to business logic. That said, I think it likely makes sense to revise `lock-game` so that it returns the game itself, which will contain the lock uuid.. Then we don't to wait for that before querying for the game because we'll get the game directly from the lock. It'll save us a query.

Then we'll wait for the game/lock before querying for the user and doing the guts of it

Reply:
@seriouslysean I've updated the `lock-game` helper to return the game object itself since the method already requires querying for it and this way consumers (like ready.js) who were going to lock the game and immediately query for it before doing other stuff (the most likely use case) can just use the lock game helper to get the game they need.

Further, I've refactored `ready.js` to use async/await so that it's easy to simply `await` the `lockGame()` call as the first line and then ensure that the rest of the data is only queried and considered valid after the lock is obtained. This should prevent weird cases around other data being queried before the lock is obtained then updated while this request is midflight